zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -226,6 +271,87 @@ where
         }
         .boxed()
     }
+
+    fn submit_block(
+        &self,
+        hex_data: String,
+        _options: Option<submit_block::JsonParameters>,
+    ) -> BoxFuture<Result<submit_block::Response>> {
+        let mut chain_verifier = self.chain_verifier.clone();
+
+        async move {
+            let block = hex::decode(hex_data).map_err(|error| Error {
+                code: ErrorCode::ServerError(0),
+                message: format!("failed to decode hexdata, error msg: {error}"),
+                data: None,
+            })?;
+
+            let block: Block = block.zcash_deserialize_into().map_err(|error| Error {
+                code: ErrorCode::ServerError(0),
+                message: format!("failed to deserialize into block, error msg: {error}"),
+                data: None,
+            })?;
```

Comment:
These should probably be "invalid" responses, because the block data is wrong.

Reply:
Updated in [`52ae062`](https://github.com/ZcashFoundation/zebra/pull/5526/commits/52ae062780bf8e033dd152dec0db33410e658ac9)